ndcube/wcs/wrappers/compound_wcs.py

```diff
@@ -0,0 +1,213 @@
+from functools import reduce
+
+import numpy as np
+from astropy.wcs.wcsapi.wrappers.base import BaseWCSWrapper
+
+__all__ = ['CompoundLowLevelWCS']
+
+
+def tuplesum(lists):
+    return reduce(tuple.__add__, map(tuple, lists))
+
+
+class Mapping:
+    """
+    Allows inputs to be reordered, duplicated or dropped.
+
+    This is a very stripped down version of `astropy.modeling.models.Mapping`
+    to be able to handle input of arbitrary type.
+
+    Parameters
+    ----------
+    mapping : tuple
+        A tuple of integers representing indices of the inputs to this model
+        to return and in what order to return them.  See
+        :ref:`compound-model-mappings` for more details.
+
+    """
+    def __init__(self, mapping):
+        self.mapping = mapping
+        self.n_inputs = max(mapping) + 1
+        self.n_outputs = len(mapping)
+
+    def __call__(self, *values):
+        return tuple(values[idx] for idx in self.mapping)
+
+    @property
+    def inverse(self):
+        mapping = tuple(self.mapping.index(idx)
+                        for idx in range(self.n_inputs))
+        return type(self)(mapping)
+
+    def __repr__(self):
+        return f'<Mapping({self.mapping})>'
+
+
+class CompoundLowLevelWCS(BaseWCSWrapper):
```

Comment:
Can this be used to solve the `NDCubeSequence` plotting/collapsing a sequence into a single cube?

Reply:
This would allow the one FITS WCS + lookup table to work (which is sorta how we use it in Extra Coords) but it wouldn't enable the lookup table of WCSes to work.